src/utils.ts

```diff
@@ -148,3 +148,10 @@ export function tsIntersectionOf(types: string[]): string {
 export function tsUnionOf(types: string[]): string {
   return types.join(" | ");
 }
+
+/**
+ * Conditionally convert value into nullable
+ */
+export function nullableWrap(nullable: boolean, value: string): string {
+  return nullable ? tsUnionOf([`(${value})`, "null"]) : value;
```

Comment:
So you know what? I realized the difficulty you’re experiencing, and it’s actually really challenging to pull off.

To start, the only thing your code missing is this:

```
-  return nullable ? tsUnionOf([`(${value})`, "null"]) : value;
+  return nullable ? tsUnionOf([`(${value})`, escape("null")]) : value;
```

But the problem you’re experiencing with the syntax error is a flaw with how this library transforms things through JSON. Normally, the library works because it transforms into a JSON-friendly syntax. But with the `(${value))` line, the JSON syntax is lost in a string interpolation.

So I’ve been thinking of a way to make this simpler, because the `escape()` / `unescape()` were “cheats” to fix the small things that broke the JSON syntax. But with this nullable edition, I think I’ll have to rethink how this library transforms things (which I’ll have to do with comments, as well).

Reply:
But what you’ve done is perfect, and I apologize that there’s a deeper issue preventing this from being merged.

Let me spend a little time thinking about this.

Reply:
Thank you for your comment and no worries. I've attempted to apply the change that you suggested but the test still fails. The reason being is the type generated like so

```ts
interface schemas {
  links?: ({\"data\":components['schemas']['file_link'][],\"has_more\":boolean,\"object\":'list',\"url\":string}) | null
//         ^
}
```

You can see that it is trying to escape the quotes for the property names which breaks prettier.

Reply:
Right, that’s from the problem I outlined. All those stringified property names get transformed back TS if they’re in JSON format with `JSON.parse()`, but they cant when wrapped with parentheses.

I think I have a solution which involves rewriting part of the library a bit, unfortunately. But I had a feeling that this was necessary because I was struggling with the same thing with JSDoc comments and `description`